x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/security/support/StringMatcherTests.java

```diff
@@ -48,10 +49,11 @@ public void testUnicodeWildcard() throws Exception {
         final String prefix = randomValueOtherThanMany(StringMatcherTests::hasHighSurrogate,
             () -> randomRealisticUnicodeOfLengthBetween(3, 5));
         final StringMatcher matcher = StringMatcher.of(prefix + "*");
+        final String normalisedPrefix = prefix.replace("\\", "");
```

Comment:
I would just push this up into the `otherThanMany` check.
 
```
randomValueOtherThanMany(StringMatcherTests::hasHighSurrogateOrBackslash
```

Reply:
Updated as suggested. 
PS: I think we do need some tests for the behaviour of the escape char. This is not the right place for it. Maybe when fixing #69851.